app/models/post.rb

```diff
@@ -1,3 +1,14 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  has_many :comments
+  has_many :likes
+  belongs_to :author, class_name: 'User'
+  before_save :post_count
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def post_count
+    author.update(posts_counter: author.posts.all.length)
+  end
```

Comment:
- [ ] **[OPTIONAL]** I would recommend you make this method private and we can save some code here.

  ```
  private

  def post_count
    author.increment!(:posts_counter)
  end
  ```